### PR TITLE
Additionally resolve pom extension

### PR DESCRIPTION
**Pull request by donovanmuller**
Addresses #79

### CODE REVIEW

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java [hunk 1]
```diff
@@ -197,6 +211,11 @@ public RemoteRepository(final String url) {
 			this.url = url;
 		}
 
+		public RemoteRepository(final String url, final Authentication auth) {
```
Comment (ilayaperumalg) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java]: Do you use it somewhere? trying to understand the reasoning for this.

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java [hunk 2]
```diff
@@ -220,6 +239,14 @@ public void setAuth(final Authentication auth) {
 
 		private String password;
 
+		public Authentication() {
+		}
+
+		public Authentication(String username, String password) {
```
Comment (ilayaperumalg) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java]: Same here. Would like to know why do you need these explicit changes.

**Comment by ilayaperumalg** [comment]
LGTM apart from the minor comments above.

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java (hunk 1 above)
Comment (donovanmuller) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java]: I do but more just for teasting convenience. See [here](https://github.com/donovanmuller/spring-cloud-deployer-openshift/blob/master/src/test/java/org/springframework/cloud/deployer/spi/openshift/factories/BuildConfigFactoryTest.java#L95)

Comment (donovanmuller) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java]: I do but more just for testing convenience. See [here](https://github.com/donovanmuller/spring-cloud-deployer-openshift/blob/master/src/test/java/org/springframework/cloud/deployer/spi/openshift/factories/BuildConfigFactoryTest.java#L95)

File: spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java (hunk 2 above)
Comment (donovanmuller) [inline spring-cloud-deployer-resource-maven/src/main/java/org/springframework/cloud/deployer/resource/maven/MavenProperties.java]: Same reason, although I should [apply it](https://github.com/donovanmuller/spring-cloud-deployer-openshift/blob/master/src/test/java/org/springframework/cloud/deployer/spi/openshift/factories/BuildConfigFactoryTest.java#L91)

**Comment by ilayaperumalg** [comment]
LGTM, rebased and merged as fac89fe55dc4292615d3443f75b49192e51a6c55

**Pull request merged**